Add peer-read dataset

vinaykudari opened the pull request.

lhoestq commented on datasets/peer_read/README.md:
```diff
@@ -0,0 +1,143 @@
+annotations_creators:
+- expert-generated
+language_creators:
+- found
+languages:
+- en
+licenses:
+- unknown
+multilinguality:
+- monolingual
+size_categories:
+- 10K<n<100K
+source_datasets:
+- original
+task_categories:
+- text-classification
+task_ids:
+- acceptability-classification
```
```suggestion
---
annotations_creators:
- expert-generated
language_creators:
- found
languages:
- en
licenses:
- unknown
multilinguality:
- monolingual
size_categories:
- 10K<n<100K
source_datasets:
- original
task_categories:
- text-classification
task_ids:
- text-classification-other-acceptability-classification
---
```

lhoestq commented on datasets/peer_read/README.md:
```diff
@@ -0,0 +1,143 @@
+
+### Data Fields
+
+[Needs More Information]
```
Can you add a brief description of each data field please ? (name, source, title, citeRegEx, etc.)

vinaykudari replied:
Couldn't find any official desc added desc based on the data fields :)

lhoestq commented on datasets/peer_read/README.md:
```diff
@@ -0,0 +1,143 @@
+
+### Data Splits
+
+[Needs More Information]
```
```suggestion
[More Information Needed]
```
can you replace it with [More Information Needed] instead ? That's what we use in all the other dataset cards

lhoestq commented on datasets/peer_read/peer_read.py:
```diff
@@ -0,0 +1,306 @@
+
+    BUILDER_CONFIGS = [
+        datasets.BuilderConfig(
+            name="parsed_pdfs",
+            version=VERSION,
+            description="This part of my dataset covers a first domain",
+        ),
+        datasets.BuilderConfig(
+            name="reviews",
+            version=VERSION,
+            description="This part of my dataset covers a second domain",
+        ),
```
can you add descriptions of each config here ?

vinaykudari replied:
Ah sorry I missed this! Done :)

lhoestq commented on datasets/peer_read/peer_read.py:
```diff
@@ -0,0 +1,306 @@
+                    ]
+                    metadata["references"] = [] if metadata["references"] is None else metadata["references"]
+                    metadata["references"] = [
+                        {
+                            "title": reference.get("title", ""),
+                            "author": reference.get("author", []),
+                            "venue": reference.get("venue", ""),
+                            "citeRegEx": reference.get("citeRegEx", ""),
+                            "shortCiteRegEx": reference.get("shortCiteRegEx", ""),
+                            "year": reference.get("year", ""),
```
Why use `get` here ?
All these fields have missing values ?

vinaykudari replied:
Yes some of the datasets don't have some of the fields

lhoestq commented on datasets/peer_read/peer_read.py:
```diff
@@ -0,0 +1,306 @@
+                        {
+                            "heading": self._decode(section.get("heading", "")),
+                            "text": self._decode(section.get("text", "")),
+                        }
```
same here

lhoestq commented on datasets/peer_read/peer_read.py:
```diff
@@ -0,0 +1,306 @@
+                            "referenceID": self._decode(reference_mention.get("referenceID", "")),
+                            "context": self._decode(reference_mention.get("context", "")),
+                            "startOffset": self._decode(reference_mention.get("startOffset", "")),
+                            "endOffset": self._decode(reference_mention.get("endOffset", "")),
```
same here

vinaykudari replied:
Yes some of the datasets don't have some of the fields

lhoestq commented on datasets/peer_read/peer_read.py:
```diff
@@ -0,0 +1,306 @@
+                                "reviewer_confidence": str(review.get("reviewer_confidence", "")),
+                                "soundness_correctness": str(review.get("soundness_correctness", "")),
+                                "appropriateness": str(review.get("appropriateness", "")),
+                                "impact": str(review.get("impact")),
```
same here

vinaykudari replied:
Yes some of the datasets don't have some of the fields

lhoestq commented on datasets/peer_read/peer_read.py:
```diff
@@ -0,0 +1,306 @@
+
+    def _generate_examples(self, filepaths, split):
+        """ Yields examples. """
+        for id_, filepath in enumerate(filepaths):
```
```suggestion
        for id_, filepath in enumerate(sorted(filepaths)):
```
we need to make the order deterministic here

vinaykudari replied:
Sure

lhoestq commented on datasets/peer_read/peer_read.py:
```diff
@@ -0,0 +1,306 @@
+                                "impact": str(review.get("impact")),
+                            }
+                            if isinstance(review, dict)
+                            else {}
```
Here you allow to add empty reviews, is this expected ?

vinaykudari replied:
yes some of the papers don't have reviews

lhoestq replied:
For a paper that doesn't have reviews wouldn't it be better to have `"reviews": []` instead of a list with empty dictionaries in it ? Indeed here you may end up with `"reviews": [{}]`

vinaykudari replied:
Agreed :) Corrected it. @lhoestq

lhoestq commented on datasets/peer_read/peer_read.py:
```diff
@@ -0,0 +1,306 @@
+                        "title": str(data.get("title", "")),
+                        "authors": data.get("authors", [])
+                        if isinstance(data.get("authors"), list)
+                        else [data.get("authors", "")],
```
here you may allow a list with an empty string in it 
Is this expected ?

vinaykudari replied:
I probably want to have just [] if there are no authors not [""] made the correction

lhoestq reviewed (commented):
Cool thank you !

I left a few comments

vinaykudari commented:
> Cool thank you !
> 
> I left a few comments

Thank you @lhoestq  addressed your comments. Haven't changed the code but I see that tests are failing now. Do I need to rebase or something?

lhoestq commented:
The CI error is not related to your dataset and is fixed on master.
You can ignore it

lhoestq reviewed (approved):
Thanks !

Looks all good now :)
